crates/conjure_core/src/rules/base.rs

```diff
@@ -515,3 +516,37 @@ fn distribute_not_over_or(expr: &Expr, _: &Model) -> ApplicationResult {
         _ => Err(ApplicationError::RuleNotApplicable),
     }
 }
+
+#[register_rule(("Base", 8800))]
+fn negated_neq_to_eq(expr: &Expr, _: &Model) -> ApplicationResult {
+    match expr {
```

Comment:
I do that and like that, but i didn't bother editing the rules in this PR, I just moved them.

Reply:
Ok. As a corollary of the campsite rules I'd say we should make the match nicer before merging now that we noticed it.

Reply:
Fixed. Because there is not box matching, I couldn't merge this into a single match; however, I inlined the if into the inner one.

Reply:
Oh interesting. I thought rust would allow what I wrote, why does that not work? I thought it would implicitly follow the ref, though I may be missing something.

Reply:
@niklasdewally tagging for visibility

Reply:
That reference is a box not a borrow. We would need to do something like this, which is nightly only: https://doc.rust-lang.org/beta/unstable-book/language-features/box-patterns.html

Reply:
I could sware I did this in some other code I wrote, but turns out you are right. thanks for checking.